### PR TITLE
Create user school infos table

### PR DESCRIPTION
**Pull request by nkiruka**
This PR is to create a new model and table, "user school infos".  The user_school_infos table will store the history of schools a teacher has provided.  

[Spec](https://docs.google.com/document/d/1pvyCDwdiyxWJsbLN83iKTtE_dlMVxKfEKTYZTjvQg98/edit)

### CODE REVIEW

File: dashboard/test/factories/user_school_infos.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# == Schema Information
```
Comment (maddiedierker) [inline dashboard/test/factories/user_school_infos.rb]: if you don't need a `:user_school_info` factory, you can probably delete this file for now. also, we have almost all of our factories in one file right now: https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/test/factories/factories.rb

not ideal, but just a heads up that when/if you do need this factory, you might want to add it there while we decide a better structure for our factories

File: dashboard/db/migrate/20190208001843_create_user_school_infos.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class CreateUserSchoolInfos < ActiveRecord::Migration[5.0]
+  def change
+    create_table :user_school_infos do |t|
+      t.integer  :user_id
```
Comment (maddiedierker) [inline dashboard/db/migrate/20190208001843_create_user_school_infos.rb]: i think you'll want to add `null: false` to some of these columns, although i'm not too familiar with what this model should look like. we should validate that this user_school_info is associated with a user and a school (i.e., add `null: false` to `user_id` and `school_info_id`), but i'm not sure about the other fields (`start_date`, `end_date`, and `last_confirmation_date`) and whether they should be mandatory.

once you make these columns required, you should also add validations to the model like:
```
validates_presence_of :user
validates_presence_of :school_info
...
```

Comment (bethanyaconnor) [reply dashboard/db/migrate/20190208001843_create_user_school_infos.rb]: I believe end_date can be null but the rest shouldn't be.

**Review by maddiedierker — APPROVED** [review]
don't forget to add presence validations (`validates_presence_of`) to your model! everything else looks great 🎉

**Comment by nkiruka** [comment]
@madelynkasula Validations added!  Thanks

File: dashboard/app/models/user_school_info.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+#  start_date             :datetime         not null
+#  end_date               :datetime
+#  school_info_id         :integer          not null
+#  last_confirmation_date :datetime         not null
+#  created_at             :datetime         not null
+#  updated_at             :datetime         not null
+#
+
+class UserSchoolInfo < ApplicationRecord
+  validates_presence_of :user
```
Comment (maddiedierker) [inline dashboard/app/models/user_school_info.rb:16]: you'll want to validate the presence of any column that cannot be null here, not just user and school_info:

```
validates_presence_of :user, :school_info, :start_date, :last_confirmation_date
```

Comment (bencodeorg) [reply dashboard/app/models/user_school_info.rb:16]: Stumbled upon this looking for something else...out of curiosity, what will happen here if a record is attempted to be added without `:school_info, :start_date, :last_confirmation_date`?

Comment (nkiruka) [reply dashboard/app/models/user_school_info.rb:16]: @bencodeorg Good question.  The record will not be added and a database error will be thrown.  I checked the users table and see that null is acceptable for the following columns :school_info, :start_date, :last_confirmation_date columns.  

In addition, I can also create a new teacher account without entering school information.  Do we want the user_school_infos table to have the same set up?

Comment (bencodeorg) [reply dashboard/app/models/user_school_info.rb:16]: I think your set up is correct -- having a null `school_info_id` in `users` is still ok (if a teacher didn't provide any school information), but we wouldn't want any records in this table to have a null `school_info_id`.

Comment (maddiedierker) [reply dashboard/app/models/user_school_info.rb:16]: If `:school_info, :start_date, :last_confirmation_date` can all be null, you will need to add another migration that makes those columns `null: true`. Right now, the database is requiring those fields via `null: false`

Comment (maddiedierker) [reply dashboard/app/models/user_school_info.rb:16]: I talked to Ben and I redact my last comment! Ignore me. @nkiruka can you create a second PR that adds the missing validations to our `UserSchoolInfo` model? We're still missing validations for `start_date` and `last_confirmation_date`

**Pull request merged**